great_expectations/core/http_handler.py

```diff
@@ -0,0 +1,77 @@
+from typing import Callable, Optional
```

Comment:
I'm not sure what this gives us over the requests library (or alternatives). We could have a helper method that returns a configured requests [Sessions object](https://requests.readthedocs.io/en/latest/user/advanced/#session-objects) with headers and timeout configurable with defaults set to what you have here. It seems like less to maintain and more fully featured.